RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc

```diff
@@ -554,12 +555,32 @@ void PulseShapeFitOOTPileupCorrection::phase1Apply(const HBHEChannelInfo& channe
       noiseDCArr[ip] = psfPtr_->getSiPMDarkCurrent(channelData.darkCurrent(),channelData.fcByPE(),channelData.lambda());
     }
 
+    // Photo statistics uncertainties
+    //      sigmaFC/FC = 1/sqrt(Ne);
+    noisePHArr[ip] = 0;
+    if(channelData.hasTimeInfo() && (charge-ped)>channelData.tsPedestalWidth(ip)) {
+      noisePHArr[ip] = (charge-ped)/sqrt((charge-ped)/channelData.fcByPE());
+    }
+
+    /*
+    // FIXME: in the future switch the photostatistics on also for the HPD
+    if(!channelData.hasTimeInfo() && (charge-ped)>channelData.tsPedestalWidth(ip)) {
+
+      // Note1. the fcByPE is not in the DB, but is hard coded in python
+      // https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_0/SimCalorimetry/HcalSimProducers/python/hcalSimParameters_cfi.py#L62
+
+      // Note2. (from kPedro): the output number of photoelectrons after smearing is treated very differently for SiPMs: *each* pe is assigned a different time based on a random generation from the Y11 pulse plus the SimHit time. In HPDs, the overall pulse is shaped all at once using just the SimHit time.
+
+      noisePHArr[ip] = (charge-ped)/sqrt(0.3305*16);
```

Comment:
@igv4321,  @kpedro88   (including Igor and Kevin in the thread)

Just would like to mention that this is supposed to be ieta-dependent HB constant/factor. That's what there is an array of 16 in the  photoelectronsToAnalog = cms.vdouble([0.3305]*16) 
So an actual constant if 0.3305 (sic).  

Currently this HB constant(s) is in _SIM_ config, which must not be accessed by RECO, so as Slava mentioned, it should be somewhere on a "neutral" ground (Calibcalorimetry/HcalPrugins)
So. I'd suggest to make a small config in 
http://cmslxr.fnal.gov/source/CalibCalorimetry/HcalPlugins/python/
accessible by both SinCalorimetry/HcalProducers and Reconstruction